lib/SNSEventEmitter.js

```diff
@@ -47,11 +47,15 @@ SNSEventEmitter.prototype.poll = function(callback){
         }, function(err, data) {
             if (err) return callback(err);
             var handles = null;
-            if (data.Messages) {            
+            if (data.Messages) {
                 // fire events
                 handles = data.Messages.map(function(msg) {
                     var content = JSON.parse(msg.Body);
                     var args = JSON.parse(content.Message);
+                    if (!util.isArray(args)) {
+                      args = [ args ];
+                    }
+                    args.unshift(content.Subject);
```

Comment:
I'm not sure this will be backwards compatible with any state currently stored in SQS queues using the previous format, which could lead to some mangled queue items during upgrades.